lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -4,20 +4,84 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
-
 #include "hermes/Platform/Intl/PlatformIntl.h"
 
 #import <Foundation/Foundation.h>
 
+NSString *u16StringToNSString(std::u16string src) {
+  auto size = src.size();
+  const auto *cString = (const unichar *)src.c_str();
+  return [NSString stringWithCharacters:cString length:size];
+}
+
+std::u16string nsStringToU16String(NSString *src) {
+  auto size = src.length;
+  std::u16string result;
+  result.resize(size);
+  [src getCharacters:(unichar *)&result[0] range:NSMakeRange(0, size)];
+  return result;
+}
+
 namespace hermes {
 namespace platform_intl {
 
```

Comment:
We should wrap these helper functions in an anonymous namespace. You can read more about that [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf22-use-an-unnamed-anonymous-namespace-for-all-internalnon-exported-entities).

You can see how helpers are wrapped in `PlatformIntlAndroid.cpp` as an example ([link](https://github.com/facebook/hermes/blob/5be5dcd5799e2a3f7bb507d9233c8548fb9f21e8/lib/Platform/Intl/PlatformIntlAndroid.cpp#L24)).

```suggestion
namespace hermes {
namespace platform_intl {

namespace {
NSString *u16StringToNSString(std::u16string src) {
  auto size = src.size();
  const auto *cString = (const unichar *)src.c_str();
  return [NSString stringWithCharacters:cString length:size];
}

std::u16string nsStringToU16String(NSString *src) {
  auto size = src.length;
  std::u16string result;
  result.resize(size);
  [src getCharacters:(unichar *)&result[0] range:NSMakeRange(0, size)];
  return result;
}
}

```